lib/task.js

```diff
@@ -665,6 +668,24 @@ export default class Task {
       return await this.abortRun(this.taskState);
     }
     this.runtime.log('task run');
+
+    // Call started hook when container is started
+    dockerProc.once('container start', async (container) => {
+      try {
+        await this.states.started(this);
+      }
+      catch (e) {
+        debug(e.stack);
+        return await this.abortRun(
```

Comment:
Do we have a test verifying that the run is aborted and marked as failure if the right conditions are present?  Could do this by changing the startup configuration of the worker to have an invalid certificate, or move the port range to the configuration and then you can specify an invalid port range (or a known used port) as part fo the startup config to make sure it fails gracefully.